File: app/src/main/java/com/dmi/perfectreader/book/animation/PageAnimationState.java
package com.dmi.perfectreader.book.animation;

import com.carrotsearch.hppc.FloatArrayList;
import com.carrotsearch.hppc.IntArrayList;

public class PageAnimationState {
    private final IntArrayList relativeIndices = new IntArrayList();
    private final FloatArrayList xPositions = new FloatArrayList();
    private int minRelativeIndex = 0;
    private int maxRelativeIndex = 0;

    void clear() {
        relativeIndices.clear();
        xPositions.clear();
        minRelativeIndex = 0;
        maxRelativeIndex = 0;
    }

    void add(int relativeIndex, float xPosition) {
        relativeIndices.add(relativeIndex);
        xPositions.add(xPosition);
        if (relativeIndex < minRelativeIndex) {
            minRelativeIndex = relativeIndex;
        }
        if (relativeIndex > maxRelativeIndex) {
            maxRelativeIndex = relativeIndex;
        }
    }

    public int pageCount() {
        return relativeIndices.size();
    }

    public int pageRelativeIndex(int index) {
        return relativeIndices.get(index);
    }

    public float pagePositionX(int index) {
        return xPositions.get(index);
    }

    public int minRelativeIndex() {
        return minRelativeIndex;
    }

    public int maxRelativeIndex() {
        return maxRelativeIndex;
    }
}

File: app/src/main/kotlin/com/dmi/perfectreader/book/animation/PageAnimationState.kt
package com.dmi.perfectreader.book.animation

import com.carrotsearch.hppc.FloatArrayList
import com.carrotsearch.hppc.IntArrayList

class PageAnimationState {
    private val relativeIndices = IntArrayList()
    private val xPositions = FloatArrayList()
    private var minRelativeIndex = 0
    private var maxRelativeIndex = 0

    fun clear() {
        relativeIndices.clear()
        xPositions.clear()
        minRelativeIndex = 0
        maxRelativeIndex = 0
    }

    fun add(relativeIndex: Int, xPosition: Float) {
        relativeIndices.add(relativeIndex)
        xPositions.add(xPosition)
        if (relativeIndex < minRelativeIndex) {
            minRelativeIndex = relativeIndex
        }
        if (relativeIndex > maxRelativeIndex) {
            maxRelativeIndex = relativeIndex
        }
    }

    fun pageCount(): Int {
        return relativeIndices.size()
    }

    fun pageRelativeIndex(index: Int): Int {
        return relativeIndices.get(index)
    }

    fun pagePositionX(index: Int): Float {
        return xPositions.get(index)
    }

    fun minRelativeIndex(): Int {
        return minRelativeIndex
    }

    fun maxRelativeIndex(): Int {
        return maxRelativeIndex
    }
}